### DISCUSSION TITLE
Why is Micronaut AOT enabled per default in newly generated applications using the Micronaut CLI?

### DISCUSSION THREAD

**Discussion by runenielsen** [micronaut-projects/micronaut-core#9826] [Q&A] [answered]
I'm wondering why Micronaut AOT is enabled per default in newly generated Micronaut 4 projects, when the [documentation](https://micronaut-projects.github.io/micronaut-aot/latest/guide/) says it is an experimental project and using it is at your own risk?

Is the project no longer experimental, and the documentation just not up-to-date, or?

Also - this is how the `aot` configuration looks in build.gradle in a new project:

```
    aot {
    // Please review carefully the optimizations enabled below
    // Check https://micronaut-projects.github.io/micronaut-aot/latest/guide/ for more details
        optimizeServiceLoading = false
        convertYamlToJava = false
        precomputeOperations = true
        cacheEnvironment = true
        optimizeClassLoading = true
        deduceEnvironment = true
        optimizeNetty = true
    }
```

But I can't find any description of the different features in the documentation main page or the [configuration reference](https://micronaut-projects.github.io/micronaut-aot/latest/guide/configurationreference.html) (which is empty).

Thanks in advance for enlightening me 🙂

**Comment by yawkat** [comment, marked as answer]
@melix 

fwiw the options are described a bit better in https://micronaut-projects.github.io/micronaut-gradle-plugin/latest/#_micronaut_aot_plugin . and the aot integration will only run if you use on the 'optimized' targets (e.g. nativeOptimizedCompile instead of nativeCompile)

**Reply by melix** [reply]
There is a [PR](https://github.com/micronaut-projects/micronaut-aot/pull/242) which adds some "configuration reference" to the AOT project, but one has to understand that these are not traditional configuration properties, but properties of the optimizer, which need to be configured in the build plugins. The docs of the Gradle plugin are indeed the place to look for.

**Reply by yawkat** [reply]
@melix can you comment on the experimental-ness?

**Reply by melix** [reply]
That comment should be removed.

**Reply by runenielsen** [reply]
@yawkat @melix 

Thanks. It makes sense, that the properties are documented in the Gradle plugin, but it is a bit confusing, that the link in the generated comment in build.gradle points to the main documentation, where you can't find any description of the properties. But it looks like the PR fixes this, thanks a lot for that.

I read Melix' comment above as AOT not being experimental anymore. In that case it makes more sense to me, that it is enabled per default, as long as the comment on it being experimental is removed 🙂
